### PR TITLE
jsc.suchthat should be blessed to support smap.

### PR DESCRIPTION
**Pull request by akiellor**

### CODE REVIEW

**Comment by phadej** [comment]
please fix `eslint` complaints.

File: test/suchthat.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/* jshint node:true */
+/* global describe, it */
+"use strict";
+
+var jsc = require("../lib/jsverify.js");
+var arbitraryAssert = require("../lib/arbitraryAssert.js");
+
+describe('suchthat', function () {
```
Comment (phadej) [inline test/suchthat.js]: double quite

File: test/suchthat.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  const arb = jsc.suchthat(jsc.integer, function (v) {
```
Comment (phadej) [inline test/suchthat.js]: we write es5 code still.

File: lib/suchthat.js [hunk 3]
```diff
@@ -43,8 +44,8 @@ function suchthat(arb, userenv, predicate) {
       return arb.shrink(x).filter(predicate);
     }),
 
-    show: arb.show,
-  };
+    show: arb.show
```
Comment (phadej) [inline lib/suchthat.js]: please leave the trailing comma (even `eslint` doesn't seem to complain about that)

**Comment by phadej** [comment]
Thanks!

**Comment by akiellor** [comment]
No worries!

**Pull request merged**